system-int-test/test/cmr/system_int_test/search/granule_spatial_search_test.clj

```diff
@@ -49,6 +51,86 @@
   [& ords]
   (codec/url-encode (umm-s/set-coordinate-system :geodetic (apply polygon ords))))
 
+;; Tests all granule search when allow-all-granule-params-flag is false
+;; Existing tests should cover the case when the flag is set to true. 
+(deftest allow-all-granule-params-flag-false-search-test 
+  (let [saved-flag-value (concepts-search/allow-all-granule-params-flag)
+        saved-header-value (concepts-search/allow-all-gran-header) 
+        _ (side/eval-form `(concepts-search/set-allow-all-granule-params-flag! false))
+        _ (side/eval-form `(concepts-search/set-allow-all-gran-header! "allow-all-gran"))
+        header1 {"client-id" "testing", "allow-all-gran" true} 
+        header2 {"client-id" "testing", "allow-all-gran" false} 
+        header3 {"client-id" "testing"}
+        header4 {"allow-all-gran" true} 
+        header5 {"allow-all-gran" false}
+        header6 {}
+        params1 {:provider "PROV1"}
+        params2 {:provider_id "PROV1"}
+        params3 {:concept_id "C1234-PROV1"}
+        params4 {:collection_concept_id "C1234-PROV1"}
```

Comment:
Can you also had tests for echo_granule_id, dataset_id, echo_collection_id, and shortName?

These are aliases we support to parameter names concept_id, entry title, collection_concept_id, and short_name, and I've seen all of these used by clients in production. They should be allowed.

Reply:
Done.

Reply:
Done.